charts/controller/templates/ingress.yaml

```diff
@@ -0,0 +1,22 @@
+{{ if .Values.ingress.enabled - }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "controller.fullname" . }}
+  labels:
+    app: {{ template "controller.name" . }}
+    chart: {{ template "controller.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  annotations:
+{{ toYaml .Values.ingress.annotations | trim | indent 4 }}
+spec:
+  rules:
+    - host: {{ .Values.ingress.host }}
```

Comment:
The host should have a default value.
Preferably the fully qualified service name (i.e.: `service_name.namespace.svc.cluster.local`)
This can be achieved the following way:

`{{ .Values.ingress.host | default "{{ template "controller.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local"}}`

Reply:
Neither in Helm2 or Helm3 does the above appear to be a valid way to define ,and always comes back with a linting error, or the literal aspects of what I'm seeing.

Reply:
Open to suggestions, but what I'm seeing from helm template etc..says that variables in default doesn't seem to work?

Reply:
Really sorry for the delay I did not get notified for some reason..
So yes I put that as an example to provide a default I did not check it I admit.

It can be achieved like so:
` - host: {{ .Values.ingress.host | default (printf "%s.%s.svc.cluster.local" (include "controller.fullname" .) .Release.Namespace) }}`
This definitely works and here's the output of helm template
![image](https://user-images.githubusercontent.com/21025364/82141504-d8263b80-983e-11ea-80fd-50e7d42c67f8.png)

Let's do that change and we'll get it approved :)
It is important for us to allow mistake-proof configuration if somebody just wants to set the ingress as enabled without thinking about anything else

Reply:
Sorry about the delay I was not notified for some reason.
Using printf works properly when setting a default value - I wasnt sure about the last snippet I posted but I did check this one :)

`- host: {{ .Values.ingress.host | default (printf "%s.%s.svc.cluster.local" (include "controller.fullname" .) .Release.Namespace) }}`

![image](https://user-images.githubusercontent.com/21025364/82141579-85994f00-983f-11ea-8d9f-fdea629afda3.png)

Let's get this change in and we'll approve and publish this version.
Thanks

Reply:
Confirmed that yep the above works. Certainly not the prettiest aspect to be able to do what is wanted, but its what we get with helm for now.